File: 303-range-sum-query-immutable/range-sum-query-immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.prefix = [0] * (len(nums) + 1)
        self.nums = nums
        for i in range(len(nums)):
            self.prefix[i + 1] = self.prefix[i] + nums[i]
        
    def sumRange(self, left: int, right: int) -> int:
        curr_sum = self.prefix[right] - self.prefix[left] + self.nums[right]
        return curr_sum

[0,-2,-2,1,-4,-2,-3]        


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)

File: 303-range-sum-query-immutable/range-sum-query-immutable.cpp
class NumArray {
private:
    vector<int> prefix;

public:
    NumArray(vector<int>& nums) {
        int n = nums.size();
        prefix.push_back(0);
        for (int n : nums) {
            prefix.push_back(prefix.back() + n);
        }
    }
    
    int sumRange(int left, int right) {
        return prefix[right + 1] - prefix[left];
    }
};
/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray* obj = new NumArray(nums);
 * int param_1 = obj->sumRange(left,right);
 */